py/README.md

```diff
@@ -0,0 +1,374 @@
+# AllOfUs Python Client Library
+
+Python code in this library is used to access the AllOfUs Workbench API from
+the context of a Jupyter notebook. 
+
+## Installing the library
+
+The latest version of the library will be installed automatically on the Jupyter
+server your notebook code runs in, but if you need to upgrade to a version which 
+has not been released yet, you can do so by running in a notebook cell:
+
+```
+!pip install --user --upgrade 'https://github.com/all-of-us/pyclient/archive/<VERSION>.zip#egg=aou_workbench_client&subdirectory=py'
+```
+
+or for Python 2.7:
+
+```
+!pip2.7 install --user --upgrade 'https://github.com/all-of-us/pyclient/archive/<VERSION>.zip#egg=aou_workbench_client&subdirectory=py'
+```
+
+and then restarting your Python kernel.
+
+The modules you will use in your code are in the aou_workbench_client package.
+
+## Using cohorts in a notebook
+
+The aou_workbench_client.cohorts module provides functions for materializing cohorts.
+
+### materialize_cohort
+
+materialize_cohort is used to fetch some or all results of cohort materialization, 
+and return a generator of dictionaries representing the results.
+
+This can be used to retrieve information about some or all of the participants in a
+cohort you defined in the AllOfUs workbench.
+
+No server call takes place to retrieve the data until you start iterating over
+the generator.
+
+For not-too-large result sets, it is reasonable to use a Pandas `DataFrame` 
+or call to `list()` to bring all the results from the generator into memory at once. 
+If the result sets are very large, you may need to stream the results to disk as you 
+use them to avoid running into out of memory errors.
+
+Note that for large cohorts, a single call to materialize_cohort may run for 
+a very long time.
+
+Full examples of calling materialize_cohort can be found [below](#putting-it-all-together).
+
+#### materialize_cohort parameters 
+Name | Required? | Description
```

Comment:
I foresee this going out of sync. I don't suppose it's possible to autogenerate Python docs here more generally? (or if this was autogenerated, please add instructions).

Reply:
Yeah, I worry about this too.

@fonnesbeck, is it useful to have the full details on all the parameters to functions and fields on objects in one place in this README? 

We already have Swagger-generated documentation on the API model objects that this README links to; we could (with some work) generate markdown from the Python code for cohorts.py, etc. as well. Is it still useful to inline this stuff here? 

If so, I agree we should think about generating them, though I'm likely to file a TODO bug to do that as it will take some work. 

If not, we can remove the details here, just link to those docs, and have less worry about things getting inconsistent. (We'd still have a problem with examples getting out of sync, so either way we're going to need to review docs every time we do a release.)